src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -578,6 +605,22 @@ object HashMap extends ImmutableMapFactory[HashMap] with BitOperations.Int {
       case hm: HashMap[_, _] => this
       case _ => sys.error("section supposed to be unreachable.")
     }
+
+    override def equals(that: Any): Boolean = {
+      that match {
+        case hm: HashTrieMap[_, _] =>
+          (this eq hm) || {
+            this.bitmap == hm.bitmap &&
+              this.size0 == hm.size0 &&
```

Comment:
I think you can skip `size0`. If I understand the code correctly, `bitCount(bitmap) == size0 == elems.length` should be an invariant.

Reply:
No, actually `size0` is the transitive size, so it makes sense to check it first.